optuna/_gp/optim_mixed.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import annotations
+
+import math
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from optuna._gp.acqf import AcquisitionFunctionParams
+from optuna._gp.acqf import eval_acqf_no_grad
+from optuna._gp.acqf import eval_acqf_with_grad
+from optuna._gp.search_space import normalize_one_param
+from optuna._gp.search_space import sample_normalized_params
+from optuna._gp.search_space import ScaleType
+from optuna.logging import get_logger
+
+
+if TYPE_CHECKING:
+    import scipy.optimize as so
+else:
+    from optuna import _LazyImport
+
+    so = _LazyImport("scipy.optimize")
+
+_logger = get_logger(__name__)
+
+
+def _local_search_continuous(
+    acqf_params: AcquisitionFunctionParams,
+    initial_params: np.ndarray,
+    initial_fval: float,
+    continuous_params: np.ndarray,
+    continuous_param_scale: np.ndarray,
+    tol: float,
+) -> tuple[np.ndarray, float] | None:
+    if len(continuous_params) == 0:
+        return None
+    normalized_params = initial_params.copy()
+
+    def negfun_continuous_with_grad(x: np.ndarray) -> tuple[float, np.ndarray]:
+        normalized_params[continuous_params] = x * continuous_param_scale
+        (fval, grad) = eval_acqf_with_grad(acqf_params, normalized_params)
+        # Flip sign because scipy minimizes functions.
```

Comment:
```suggestion
        # Flip sign because scipy minimizes functions.
        # Let the scaled acqf be g(x) and the acqf be f(sx), then dg/dx = df/dx * s.
```